packages/davinci-client/src/lib/davinci.types.ts

```diff
@@ -116,13 +116,43 @@ export type MultiSelectFieldValue = {
   type: 'CHECKBOX' | 'COMBOBOX';
 };
 
+export type DeviceAuthenticationFieldValue = {
+  type: 'DEVICE_AUTHENTICATION';
+  key: string;
+  label: string;
+  devices: {
+    type: string;
+    iconSrc: string;
+    title: string;
+    id: string;
+    default: boolean;
+    value: string;
+  }[];
+  required: boolean;
+};
+
+export type DeviceRegistrationFieldValue = {
+  type: 'DEVICE_REGISTRATION';
+  key: string;
+  label: string;
+  devices: {
+    type: string;
+    iconSrc: string;
+    title: string;
+    description: string;
+  }[];
+  required: boolean;
+};
+
 export type DaVinciField =
```

Comment:
As this "type" grows, I wonder if we should consider breaking this into some groups of subtypes.

Not blocking for this PR, and I do think this organization has pro's but also negatives as we have to think about where things go and this is exposed to the consumer, so we'd want to document this well so a consumer knows what these fields are when dealing with more "primitive" types to the group.


i.e. DeviceFields | InputFields | ValidationFields | RedirectFields (naming subject to change

Reply:
I agree with this. I'll do it now.